### PR TITLE
add getPageStatusIcon hook to api documentation

### LINKED ISSUE #285: [3.5] add getPageStatusIcon hook to api documentation

### PR DESCRIPTION
**Pull request by bezin**
Fixes #285

### CODE REVIEW

File: api/hooks/getPageStatusIcon.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	Database result set from table `tl_page`.
+
+2. *string* `$image`
+
+	The file name of the default status icon calculated by Contao.
+
+
+## Return Value ##
+
+You can specify a new image to be displayed in the backend by returning a filename.
```
Comment (lionel-m) [inline api/hooks/getPageStatusIcon.md]: backend -> back end

File: api/hooks/getPageStatusIcon.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# getPageStatusIcon #
+
+The `getPageStatusIcon` hook is triggered when the appropriate page status icon is calculated. It passes the database result object and the filename of the current icon as arguments and expects a file name as return value. 
+> **tag** It is available from version 3.5.0.
```
Comment (aschempp) [inline api/hooks/getPageStatusIcon.md]: Please write **Tag** in uppercase and add an empty line before it.

File: api/hooks/getPageStatusIcon.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+
+## Parameters ##
```
Comment (aschempp) [inline api/hooks/getPageStatusIcon.md]: Please do not add `##` at the end

File: api/hooks/getPageStatusIcon.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	Database result set from table `tl_page`.
+
+2. *string* `$image`
+
+	The file name of the default status icon calculated by Contao.
+
+
+## Return Value ##
+
+You can specify a new image to be displayed in the back end by returning a filename.
```
Comment (aschempp) [inline api/hooks/getPageStatusIcon.md]: This should describe that one must always return the desired image, either the second parameter or by changing it.

File: api/hooks/getPageStatusIcon.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+```php
+<?php
+
+// config.php
+$GLOBALS['TL_HOOKS']['myGetPageStatusIcon'][] = array('MyClass', 'myGetPageStatusIcon');
+
+// MyClass.php
+public function myGetPageStatusIcon(Database_Result $objPage, $image)
+{
+    return 'custom_icon.gif';
```
Comment (aschempp) [inline api/hooks/getPageStatusIcon.md]: I would suggest the following example code:

``` php
if ($objPage->type == 'my_page')
{
    return 'path/to/custom_icon.gif';
}

return $image;
```

File: api/hooks/getPageStatusIcon.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+// config.php
+$GLOBALS['TL_HOOKS']['myGetPageStatusIcon'][] = array('MyClass', 'myGetPageStatusIcon');
+
+// MyClass.php
+public function myGetPageStatusIcon(Database_Result $objPage, $image)
+{
+    if ($objPage->type == 'my_page')
+    {
+        return 'path/to/custom_icon.gif';
+    }
```
Comment (aschempp) [inline api/hooks/getPageStatusIcon.md:41]: we're still missing the "else" return value ;-)

File: api/hooks/getPageStatusIcon.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+}
+```
+
+
+## More Information
+
+
+### Reference
+
+- [system/libraries/Controller.php](https://github.com/contao/core/blob/master/system/modules/core/library/Contao/Controller.php#L608)
```
Comment (aschempp) [inline api/hooks/getPageStatusIcon.md]: Please always link to a specific version (see https://github.com/contao/docs/issues/311)

**Comment by aschempp** [comment]
Can you please fix the remaining small issues? :-)

**Comment by bezin** [comment]
I am very sorry, I was pretty sure I fixed all of them (even though I know I wanted to start working on the cookbook chapter about models) ;) Anyhow, here you are.

**Pull request merged**